ote/src/clj/ote/services/login.clj

```diff
@@ -103,51 +101,48 @@
            {:success? true
             :session-data
             (let [user (users/find-user db (:name login-info))]
-              (transport/get-user-transport-operators-with-services db (:groups user) (:user user)))})
+              (transport/get-user-transport-operators-with-services db (:groups user) (:user user)))}
+           200)
           (cookie/unparse "0.0.0.0" (:shared-secret auth-tkt-config)
                           {:digest-algorithm (:digest-algorithm auth-tkt-config)
                            :timestamp (java.util.Date.)
                            :user-id (:name login-info)
                            :user-data ""})
           (:domain auth-tkt-config))
 
-        ;; No need to hide if the error was in the email or the password
-        ;; the registration page can be used to check if an email has an account
-        ;; Update 8.4.2019: We decided that in login form we will only indicate error in more general way.
-        (http/transit-response {:error :login-error}))
-      (http/transit-response {:error :login-error}))))
+        ;; Login will only indicate error in a general way and not reveal which credential was invalid
+        (http/transit-response {:error :login-error} 200))
+      (http/transit-response {:error :login-error} 200))))
 
-(defn logout [auth-tkt-config]
-  (with-auth-tkt (http/transit-response :ok) "" (:domain auth-tkt-config)))
+(defn- logout [auth-tkt-config]
+  (with-auth-tkt (http/transit-response :ok 200) "" (:domain auth-tkt-config)))
 
-(defn valid-registration? [{:keys [username name email password]}]
+(defn- valid-registration? [{:keys [name email password]}]
   (and (user/password-valid? password)
        (user/email-valid? email)
-       (user/username-valid? username)
        (string? name) (not (str/blank? name))))
 
-(defn valid-user-save? [{:keys [username name email]}]
+(defn- valid-user-save? [{:keys [username name email]}]
   (and (user/email-valid? email)
        (user/username-valid? username)
        (string? name) (not (str/blank? name))))
 
-(defn- register-user! [db auth-tkt-config {:keys [username name email password] :as form-data}]
+(defn- register-user! [db auth-tkt-config {:keys [name email password] :as form-data}]
   (if-not (valid-registration? form-data)
     ;; Check errors that should have been checked on the form
     {:success? false}
     (with-transaction db
-      (let [username-taken? (username-exists? db {:username username})
-            email-taken? (email-exists? db {:email email})]
-        (if (or username-taken? email-taken?)
+      (let [email-taken? (email-exists? db {:email email})
+            user-id (str (UUID/randomUUID))]
+        (if email-taken?
           ;; Username or email taken, return errors to form
           {:success? false
-           :username-taken (when username-taken? username)
            :email-taken (when email-taken? email)}
```

Comment:
no need for (when taken) anymore because it is taken due to if above

Reply:
Fix'd